docs/requirements.md

```diff
@@ -2,88 +2,78 @@
 
-## Cluster resource requirements
+## Hardware requirements
 
-To have an enjoyable experience with Codacy, you should have the
-following calculations in mind when allocating resources for the
-installation and defining the number of concurrent analysis.
+The following high-level architecture is important in understanding how Codacy uses and allocates hardware resources.
 
-Without accounting for analysis ([next section](#analysis)),
-you should need at least the following resources:
+You can look at Codacy separately as the **"Platform"** and the **"Analysis"** cluster. The Platform contains the UI and other
+components important to treat and show results. The Analysis is the swarm of workers that run several linters for
+your project.
 
-```text
-CPU: 7 CPU
-Memory: 40 GB
-```
+**The resources needed for Codacy depend a lot on the rate of commits done by your team.**
 
-Check the
-[values-production.yaml](https://github.com/codacy/chart/blob/master/codacy/values-production.yaml)
-file to find a configuration reference that should work for you to run a
-"production-ready" version of Codacy.
+!["High Level Architecture"](<./images/High Level Architecture - Analysis II.svg> "High Level Architecture")
 
-### Analysis
+### Virtual CPUs and memory
 
-Each analysis runs a maximum number of 4 plugins in parallel (not configurable)
+Since all components are running on Kubernetes, you can increase the number of replicas in every deployment to give you more resilience and throughput, at a cost of increased resource usage. The following is a simplified overview of how to calculate resource allocation for the **"Platform"** and the **"Analysis"**:
 
-Note: All the following configurations are nested inside the `worker-manager.config`
-configuration object, but for simplicity, we decided to omit the full path.
+| Component                          | vCPU                    | Memory                   |
+| ---------------------------------- | ----------------------- | ------------------------ |
+| Platform (1 replica per component) | 4                       | 8                        |
+| Analysis Worker + 4 linters        | 5 (per Analysis Worker) | 10 (per Analysis Worker) |
 
-```text
-CPU: workerResources.requests.cpu + (pluginResources.requests.cpu * 4)
+#### Standard Cluster Installation
 
-Memory: workerResources.requests.memory + (pluginResources.requests.memory * 4)
+The resources described on the following table are based on our experience and are also the defaults in the [values-production.yaml](https://raw.githubusercontent.com/codacy/chart/master/codacy/values-production.yaml) file, which you might need to adapt taking into account your use case.
 
-Number of concurrent analysis: workers.dedicatedMax
-```
+| Installation type                        | ~ Total vCPUs | ~ Total Memory | Replicas per component | Max. commits analyzed concurrently | Platform vCPUs | Platform Memory | Analysis Workers vCPUs | Analysis Workers Memory |
```

Comment:
Totals in the rightmost columns? (just a personal preference)

Reply:
It was a suggestion from @pauloribeiro-codacy that we were trying out, but I also like it more on the right. Changing 👍